server/app/acs_enterprise/utils/dynamodb_utils.py

```
@@ -29,174 +28,196 @@ class ConcurrencyController:
+                logger.info(f"✅ Released concurrency slot for call {call_id} (active: {new_count}/{self.max_concurrent})")
+                return True
 
         except ClientError as e:
             if e.response['Error']['Code'] == 'ConditionalCheckFailedException':
                 # Counter already at 0, cannot decrement further
-                logger.warning(f"⚠️ Cannot release slot - counter already at 0 (possible double-release)")
+                logger.warning(f"⚠️ Cannot release slot for call {call_id} - counter already at 0 (possible double-release)")
                 return False
             else:
-                logger.error(f"❌ DynamoDB error releasing slot: {e}")
+                logger.error(f"❌ DynamoDB error releasing slot for call {call_id}: {e}")
```

Ruff (0.15.1)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)